sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2IndexesResourceImpl.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.restsvc.resources.schemas;
+
+import io.stargate.grpc.StargateBearerToken;
+import io.stargate.proto.QueryOuterClass.Query;
+import io.stargate.proto.Schema;
+import io.stargate.proto.StargateGrpc;
+import io.stargate.sgv2.common.cql.builder.Predicate;
+import io.stargate.sgv2.common.cql.builder.QueryBuilder;
+import io.stargate.sgv2.restsvc.grpc.BridgeSchemaClient;
+import io.stargate.sgv2.restsvc.impl.GrpcClientFactory;
+import io.stargate.sgv2.restsvc.models.Sgv2IndexAddRequest;
+import io.stargate.sgv2.restsvc.resources.ResourceBase;
+import io.stargate.sgv2.restsvc.resources.Sgv2RequestHandler;
+import java.util.Collections;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/v2/schemas/keyspaces/{keyspaceName}/tables/{tableName}/indexes")
+@Produces(MediaType.APPLICATION_JSON)
+@Singleton
+public class Sgv2IndexesResourceImpl extends ResourceBase implements Sgv2IndexesResourceApi {
+
+  @Inject private GrpcClientFactory grpcFactory;
+
+  @Override
+  public Response getAllIndexesForTable(
+      String token, String keyspaceName, String tableName, HttpServletRequest request) {
+    if (isAuthTokenInvalid(token)) {
+      return invalidTokenFailure();
+    }
+    return Sgv2RequestHandler.handleMainOperation(
+        () -> {
+          if (isStringEmpty(keyspaceName)) {
+            return jaxrsBadRequestError("keyspaceName must be provided").build();
+          }
+          if (isStringEmpty(tableName)) {
+            return jaxrsBadRequestError("tableName must be provided").build();
+          }
+
+          StargateGrpc.StargateBlockingStub blockingStub =
+              grpcFactory
+                  .constructBlockingStub()
+                  .withCallCredentials(new StargateBearerToken(token));
+
+          BridgeSchemaClient.create(blockingStub).findTable(keyspaceName, tableName);
```

Comment:
How expensive is to create stub and client on every request? I know it depends on the `token` but wander if there is better solution, or is this how the grpc would work by default?

Also, seems that this is actually gonna be a huge perf issue, let me try to explain, maybe I am wrong:
* In the `GrpcClientFactory` you construct the `ManagedChannelBuilder` and set it to use the `directExecutor`
* The Javadoc there is clear `Execute application code directly in the transport thread. Depending on the underlying transport, using a direct executor may lead to substantial performance improvements. However, it also requires the application to not block under any circumstances.`, but I am not sure how this relates to the gRPC clients
* Then you have the blocking stub in all calls, which
    * makes me think `directExecutor` should not be used (not sure)
    * will make a context switch every time you make a call.. I experienced this issue with the docs api perf tests, you have a bunch of "http" threads accepting the requests and then they block on calling the bridge, it's massive context switch all the time

I don't understand why are we not using the standard async stub here? You would just need to wire the `AsyncResponse` and have an implementation of the `StreamObserver` that actually reacts on the gRPC response, transforms to the HTTP response and pass it to the Async response.. We do something similar in the [AsyncObserver](https://github.com/stargate/stargate/blob/c44bcb22c4b6175de89f636f8fc09c8881554c30/restapi/src/main/java/io/stargate/web/docsapi/resources/async/AsyncObserver.java#L36) but for the Rx.. Note that some frameworks provide OOB support and wrappers for Stubs in order to bridge them to Rx or Reactor, meaning that it takes effort for users to actually implement and bridge HTTP async response to gRPC stream observer and vice versa..

I guess this has nothing to do with this PR but is a bigger issue with the V2.. If we are to measure the performance of the new architecture we should avoid mistakes done in V1.. And V1 is blocking all the way..

Reply:
Our first goal is to prove that functionality works, and see what performance is. Then we can focus on improvements -- schema access, for example, is probably going to be too expensive (as implemented) to be feasible now.

I think we really should get blocking variant working with limited amount of work and then figure out how to convert.
If there are simple improvements to use of blocking style those could be added first of course.

Also need to consider that JAX-RS is fully blocking so we need to make sure that async operation on gRPC access actually measurable helps. I know async approach can help with massively parallel operation, and SGv2 is proxy-like enough to fit into that pattern. But just because something is non-blocking does not automatically mean it is more efficient so need to make sure we compare things.

Reply:
> I don't understand why are we not using the standard async stub here? You would just need to wire the `AsyncResponse` and have an implementation of the `StreamObserver` that actually reacts on the gRPC response, transforms to the HTTP response and pass it to the Async response..

Agreed, I plan to look at that next.

> Also need to consider that JAX-RS is fully blocking

How so? With `@Suspended` and `AsyncResponse` it's possible to implement non-blocking resources.

Reply:
Some additional observations I made:

* the thread-per-request model can actually be fine here, as besides some basic processing of the request the threads does a call outside this JVM and blocks, which actually is fine (opposite to V1, which blocks so that other thread does the expensive CPU work in the same JVM)
* so Jetty used by Dropwizard is actually blocking, although they do support asynchronous processing.. To be fully reactive you need to have whatever `reactive-netty` or `underthrow` as the server.. mapping this to the async stub will provide benefits with to performance and resource used, I am certain of this..
* I am fine sticking with blocking until we actually move away from Dropwizard.. Implementing async processing did not showcase heavy improvements in the docs-api V1, however this can be due to the fact of how is the async support implemented in Jetty (you still get a request per thread)

Reply:
I agree with points above by @ivansenic . I believe that fully async operations can help with highly concurrent systems. I am also not against exploring async processing within current system, to see how simple/complex changes would be.
Just want to make sure we get benefits we want and decide which parts to replace in which order.

@olim7t yes, true, servlet API (and them Jersey) do have async processing model. But by default handling is blocking so changes would be needed.